### PR TITLE
Extends collection of shutdown Reasons (e.g. ClusterJoinUnsuccessfulReason)

### PR DESCRIPTION
**Pull request by ignasi35**
Related to https://github.com/lagom/lagom/issues/1806

### CODE REVIEW

File: akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala [hunk 1]
```diff
@@ -140,6 +140,27 @@ object CoordinatedShutdown extends ExtensionId[CoordinatedShutdown] with Extensi
    */
   def clusterDowningReason: Reason = ClusterDowningReason
 
+  /**
+    * Scala API: The shutdown was initiated by a failure to join a seed node.
+    */
+  object JoiningSeedNodesUnsuccessfulReason extends Reason
```
Comment (johanandren) [inline akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala]: That seems extremely specific, should it rather be `ClusterJoinUnsuccessful` or something like that, or is it important that it was specifically joining to seed nodes that failed?

File: akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala [hunk 2]
```diff
@@ -140,6 +140,27 @@ object CoordinatedShutdown extends ExtensionId[CoordinatedShutdown] with Extensi
...
+
+  /**
+    * Scala API: The shutdown was initiated by a configuration clash within the existing cluster and the joining node
+    */
+  case object IncompatibleConfigurationDetectedReason extends Reason
+
+  /**
+    * Java API: The shutdown was initiated by a configuration clash within the existing cluster and the joining node
+    */
+  def incompatibleConfigurationDetectedReason: Reason = IncompatibleConfigurationDetectedReason
```
Comment (johanandren) [inline akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala]: Sounds like it makes sense to make it public

**Review by patriknw — COMMENTED** [review]
seems good to add those

File: akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala (hunk 1 above)
Comment (patriknw) [reply akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala]: I agree, could be slightly less specific

Comment (ignasi35) [reply akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala]: SGTM

File: akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala [hunk 3]
```diff
@@ -501,7 +501,7 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
       seedNodes.mkString(", "),
       ShutdownAfterUnsuccessfulJoinSeedNodes)
     joinSeedNodesDeadline = None
-    CoordinatedShutdown(context.system).run(CoordinatedShutdown.ClusterDowningReason)
+    CoordinatedShutdown(context.system).run(CoordinatedShutdown.ClusterJoinUnsuccessfulReason)
```
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala]: Currently we have `ClusterDowningReason` configured to use exit code `-1` (abnormal termination). Shouldn't ClusterJoinUnsuccessfulReason also be configured to use `-1`.

I guess -1 will have the effect that an orchestration platform will start it again, which is probably desired also for unsuccessful join?

Comment (ignasi35) [reply akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala]: Hmmm, I think the exit `status` was relevant in ConductR but I don't think it is in Kubernetes. IDK about other orchestrators.

I'll fix this PR but I think we could consider deprecating the feature.

Comment (patriknw) [reply akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala]: The exit code is useful. k8 isn't the only thing. A bash script can check exit code for decision of restarting process.

Comment (ignasi35) [reply akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala]: Hmmm, in the case of `IncompatibleConfigurationDetectedReason` restarting won't help so I don't think a custom exit code is necessary.

Comment (patriknw) [reply akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala]: I agree, IncompatibleConfigurationDetectedReason

**Comment by ignasi35** [comment]
The Travis job failed this morning but I think the cause was a failing bintray. Can someone relaunch, please?

**Comment by johanandren** [comment]
PLS BUILD

**Review by johanandren — APPROVED** [review]

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**